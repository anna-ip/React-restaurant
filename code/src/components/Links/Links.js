import React from 'react'
import { Link } from 'react-router-dom'
import styles from './Links.module.scss'

const Links = (props) => {
  const { to, text } = props
  //console.log('text', JSON.stringify(props))
  return (
    <Link to={to} className={styles.styledLink}>
      {text}
    </Link>
  )
}

export default Links
