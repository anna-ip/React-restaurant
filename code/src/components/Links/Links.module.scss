@import '../../styles/_variables.scss';

a {
  color: $red-color;
  text-decoration: none;
  font-size: 1.2rem;

  &:hover {
    text-decoration: underline;
  }
}
