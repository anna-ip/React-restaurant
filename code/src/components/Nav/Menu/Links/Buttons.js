import React from 'react'
import { useHistory } from 'react-router-dom'
import useViewPort from '../../../../hooks/useViewport'
import styles from './Buttons.module.scss'

const Buttons = ({ open, setOpen }) => {
  const { width } = useViewPort()
  const breakpoint = 680
  const history = useHistory()

  const handleRedirect = (path) => {
    width < breakpoint ? setOpen(!open) : setOpen(open)
    history.push(`${path}`)
  }

  return (
    <>
      <button className={styles.styledLink} onClick={() => handleRedirect('/')}>
        Home
      </button>
      <button
        className={styles.styledLink}
        onClick={() => handleRedirect('/menu')}
      >
        Menu
      </button>
      <button className={styles.styledLink} onClick={() => handleRedirect('/')}>
        About
      </button>
      <button className={styles.styledLink} onClick={() => handleRedirect('/')}>
        Contact
      </button>
    </>
  )
}

export default Buttons
