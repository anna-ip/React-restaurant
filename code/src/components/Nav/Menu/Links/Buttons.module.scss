@import '../../../../styles/variables';
@import '../../../../styles/mixins';

.styledLink {
  cursor: pointer;
  color: $red-color;
  text-decoration: none;
  border: none;
  font-size: 1.2rem;
  padding: 0 0.8rem;
  background: transparent;

  &:hover {
    text-decoration: underline;
  }

  @media (max-width: $layout-breakpoint-small) {
    text-transform: uppercase;
    font-size: 2rem;
    padding: 2rem 0;
  }
}
