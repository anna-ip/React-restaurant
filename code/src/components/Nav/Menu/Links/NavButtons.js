import React from "react";
import NavLinks from "../../../../ui-components/NavLink/NavLinks";
import styles from "./NavButtons.module.scss";

const NavButtons = () => {
  return (
    <>
      <NavLinks to="/" text="Home" className={styles.styledLink} />
      <NavLinks to="/menu" text="Menu" className={styles.styledLink} />
      <NavLinks to="/" text="About" className={styles.styledLink} />
      <NavLinks to="/" text="Contact" className={styles.styledLink} />
    </>
  );
};

export default NavButtons;
