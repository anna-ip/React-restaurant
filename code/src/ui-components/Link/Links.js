import React from "react";
import { Link } from "react-router-dom";
import styles from "./Links.module.scss";

const Links = ({ to, text, className }) => {
  return (
    <Link to={to} text={text} className={className + " " + styles.link}>
      {text}
    </Link>
  );
};

export default Links;
