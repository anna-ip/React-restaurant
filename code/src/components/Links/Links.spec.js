import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Links from './Links.js'
import { shallow } from 'enzyme'

//mock the react-router to be able to snapshot the Link component
jest.mock('react-router-dom')

describe('<Links/>', () => {
  const props = {
    to: '/menu',
    text: 'Take me to the',
  }

  test('<Link /> renders "null"', () => {
    //not sure this is the correct way?
    Router.mockReturnValue({
      Router: () => {},
    })

    const tree = renderer
      .create(<Links to="/menu">Take me to the</Links>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Render correctly', () => {
    const wrapper = shallow(<Links props={props} />)
    expect(wrapper).toMatchSnapshot()
  })

  // test render props?
})
