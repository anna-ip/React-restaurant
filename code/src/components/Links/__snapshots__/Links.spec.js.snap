// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links/> <Link /> renders "null" 1`] = `null`;

exports[`<Links/> Render correctly 1`] = `
<ForwardRef(Link)
  className={null}
/>
`;
